### PR TITLE
Added snapScheduleParameters in the modify method 

### PR DESCRIPTION
**Pull request by meenakshidembi691**
Added snapScheduleParameters in the modify method so that snap schedule can be edited via playbook

### CODE REVIEW

File: storops/unity/resource/filesystem.py [hunk 1]
```diff
@@ -103,6 +103,8 @@ def modify(self, size=None, is_thin=None, tiering_policy=None,
             params['cifsFsParameters'] = cifs_fs_parameters
         if description is not None:
             params['description'] = description
+        if snapScheduleParameters:
```
Comment (yong-huang) [inline storops/unity/resource/filesystem.py:106]: Need unit tests to cover this change.

**Review by yong-huang — CHANGES_REQUESTED** [review]

File: storops/unity/resource/filesystem.py (hunk 1 above)
Comment (meenakshidembi691) [reply storops/unity/resource/filesystem.py:106]: Have added the test for snapschedule parameter

File: storops_test/unity/resource/test_filesystem.py [hunk 2]
```diff
@@ -245,6 +245,16 @@ def test_modify_success_empty_description(self):
...
 
+    @patch_rest
+    def test_modify_success_snap_schedule(self):
+        fs = UnityFileSystem(cli=t_rest(), _id='fs_22')
+
+        snapScheduleParameters = {'snapSchedule': {'id': 'snapSch_1'}, 'isSnapSchedulePaused': False}
+
+        fs.modify(snapScheduleParameters=snapScheduleParameters)
+        fs_snapSchedule = fs.snapScheduleParameters['snapSchedule']
+        assert_that(fs_snapSchedule['id'], equal_to('snapSch_1'))
```
Comment (yong-huang) [inline storops_test/unity/resource/test_filesystem.py:256]: Test failed with:

```
E           MockFileNotFoundError: cannot find response for url: /api/instances/storageResource/res_22/action/modifyFilesystem?compact=True, 
E           body: {u'snapScheduleParameters': {u'isSnapSchedulePaused': False, u'snapSchedule': {u'id': u'snapSch_1'}}}
storops_test/unity/rest_mock.py:99: MockFileNotFoundError
```

Need to update index.json in `storops_test/unity/rest_data/storageResource/`

**Review by yong-huang — CHANGES_REQUESTED** [review]

File: storops_test/unity/resource/test_filesystem.py (hunk 2 above)
Comment (meenakshidembi691) [reply storops_test/unity/resource/test_filesystem.py]: Done

File: storops/unity/resource/filesystem.py [hunk 3]
```diff
@@ -80,7 +80,7 @@ def create(cls, cli, pool, nas_server, name, size, proto=None,
     def modify(self, size=None, is_thin=None, tiering_policy=None,
                user_cap=False, is_compression=None, access_policy=None,
                locking_policy=None, description=None,
-               cifs_fs_parameters=None):
+               cifs_fs_parameters=None, snapScheduleParameters=None):
```
Comment (yong-huang) [inline storops/unity/resource/filesystem.py:83]: snapScheduleParameters need to change to snap_schedule_parameters

File: storops/unity/resource/filesystem.py (hunk 1 above)
Comment (yong-huang) [inline storops/unity/resource/filesystem.py:106]: snapScheduleParameters need to change to snap_schedule_parameters

File: storops/unity/resource/filesystem.py [hunk 4]
```diff
@@ -103,6 +103,8 @@ def modify(self, size=None, is_thin=None, tiering_policy=None,
             params['cifsFsParameters'] = cifs_fs_parameters
         if description is not None:
             params['description'] = description
+        if snapScheduleParameters:
+            params['snapScheduleParameters'] = snapScheduleParameters
```
Comment (yong-huang) [inline storops/unity/resource/filesystem.py:107]: The second snapScheduleParameters need to change to snap_schedule_parameters

File: storops_test/unity/resource/test_filesystem.py [hunk 5]
```diff
@@ -245,6 +245,13 @@ def test_modify_success_empty_description(self):
         fs.modify(description='')
         assert_that(fs.description, equal_to(''))
 
+    @patch_rest
+    def test_modify_success_snap_schedule(self):
+        fs = UnityFileSystem(cli=t_rest(), _id='fs_22')
+        snapScheduleParameters = {'snapSchedule': {'id': 'snapSch_1'}}
```
Comment (yong-huang) [inline storops_test/unity/resource/test_filesystem.py:251]: snapScheduleParameters need to change to snap_schedule_parameters

File: storops_test/unity/resource/test_filesystem.py [hunk 6]
```diff
@@ -245,6 +245,13 @@ def test_modify_success_empty_description(self):
...
+        resp = fs.modify(snapScheduleParameters=snapScheduleParameters)
```
Comment (yong-huang) [inline storops_test/unity/resource/test_filesystem.py:252]: snapScheduleParameters need to change to snap_schedule_parameters

File: storops/unity/resource/filesystem.py (hunk 3 above)
Comment (meenakshidembi691) [reply storops/unity/resource/filesystem.py:83]: Done

File: storops/unity/resource/filesystem.py (hunk 1 above)
Comment (meenakshidembi691) [reply storops/unity/resource/filesystem.py:106]: Done

File: storops/unity/resource/filesystem.py (hunk 4 above)
Comment (meenakshidembi691) [reply storops/unity/resource/filesystem.py:107]: Done

File: storops_test/unity/resource/test_filesystem.py (hunk 5 above)
Comment (meenakshidembi691) [reply storops_test/unity/resource/test_filesystem.py:251]: Done

File: storops_test/unity/resource/test_filesystem.py (hunk 6 above)
Comment (meenakshidembi691) [reply storops_test/unity/resource/test_filesystem.py:252]: Done

File: storops/unity/resource/filesystem.py (hunk 4 above)
Comment (meenakshidembi691) [reply storops/unity/resource/filesystem.py]: Done

File: storops_test/unity/resource/test_filesystem.py (hunk 6 above)
Comment (meenakshidembi691) [reply storops_test/unity/resource/test_filesystem.py]: Done

File: storops_test/unity/resource/test_filesystem.py (hunk 5 above)
Comment (meenakshidembi691) [reply storops_test/unity/resource/test_filesystem.py]: Done

File: storops/unity/resource/filesystem.py (hunk 3 above)
Comment (meenakshidembi691) [reply storops/unity/resource/filesystem.py]: Done

**Review by yong-huang — APPROVED** [review]

**Pull request merged**